Eshop/apps/carts/views.py

```diff
@@ -3,25 +3,24 @@
 from rest_framework import generics
 from django.shortcuts import get_object_or_404
 from apps.carts.serializers import CartSerializer, CartItemSerializer
-
+from apps.carts.permissions import IsOwnerOrReadOnly
 
 def index(request):
     cart_list = CartItem.objects.all()
     return HttpResponse(', '.join([c.CartItem for c in cart_list]))
 
-
-class CartInfo(generics.ListCreateAPIView):
+class CartDetail(generics.RetrieveUpdateDestroyAPIView):
     
     queryset = Cart.objects.all()
     serializer_class = CartSerializer
+    permission_classes = (IsOwnerOrReadOnly, )
 
-
-
-class CartDetail(generics.RetrieveUpdateDestroyAPIView):
+    def get_object(self):
+        obj = get_object_or_404(CartDetail.queryset, pk=self.kwargs.get('pk'))
+        return obj
     
-    queryset = CartItem.objects.all()
-    serializer_class = CartItemSerializer
+class CartItemCreate(generics.CreateAPIView):
 
-    def get_object(self):
-        obj = get_object_or_404(CartDetail.queryset, pk=self.kwargs.get('cart_item_id'))
-        return obj
\ No newline at end of file
+    serializer_class = CartItemSerializer
+    permission_classes = (IsOwnerOrReadOnly, )
```

Comment:
permission_classes = (
        IsAuthenticated,
        Or(
            And(
                RequestIsReadOnly,
                ProfileIsVisibleToUser
            ),
            And(
                RequestIsCreate,
                ProfileIsEditableByUser,
                ProfileIsAllowedRelations
            )
        )
    )